### DISCUSSION TITLE
Headless is slower than headed

### DISCUSSION THREAD

**Discussion by TheColorRed** [cypress-io/cypress#20449] [CI setup] [answered]
I have viewed multiple issues regarding the slowness of "headed vs headless", however everything seems to point towards the issue being between test files, but that doesn't seem like the case for me. I have one file and the speed differences are for each step.

When I run through `cypress open`, I get a total runtime of `~16 seconds` as seen by this screenshot:

```
cypress run --spec **/accordian.feature --headed --browser=electron
```

![Screen Shot 2022-03-02 at 10 57 51 AM](https://user-images.githubusercontent.com/1571806/156415034-722cab46-f7b3-4d7e-b95c-870a98c4b320.png)

When running headless, I get a total runtime of `~50 seconds`.

```
cypress run --spec **/accordian.feature --headless --browser=electron
```

<img width="808" alt="Screen Shot 2022-03-02 at 11 02 15 AM" src="https://user-images.githubusercontent.com/1571806/156415244-19a2e11b-ad95-4166-97b3-3930da56733e.png">
 
Why is there such as huge difference in the time it takes to run each step in the file?

**Comment by TheColorRed** [comment, marked as answer]
I just realized that my CI tests take a snapshot whereas my headed does not, this is causing the slowness.

**Reply by ozkanpakdil** [reply]
sorry for my ignorance, can you tell how to enable disable snapshots ? I tried google but it bring screenshot related results :)

**Reply by kaisteel** [reply]
@ozkanpakdil I think you can set `screenshotOnRunFailure` to `false` and also `video` to `false` in the cypress-config.js file as mentioned [here.](https://docs.cypress.io/guides/guides/screenshots-and-videos#:~:text=Screenshots%20on%20failure%20are%20not,defaults().)

**Reply by smngvlkz** [reply]
> @ozkanpakdil I think you can set `screenshotOnRunFailure` to `false` and also `video` to `false` in the cypress-config.js file as mentioned [here.](https://docs.cypress.io/guides/guides/screenshots-and-videos#:~:text=Screenshots%20on%20failure%20are%20not,defaults().)

But will this work? Has anyone confirmed if this worked for them? If not, what did?
